src/api/schemas/wallet.ts

```diff
@@ -0,0 +1,48 @@
+/**
+ * Copyright (c) Hathor Labs and its affiliates.
+ *
+ * This source code is licensed under the MIT license found in the
+ * LICENSE file in the root directory of this source tree.
+ */
+
+import { z } from 'zod';
+import { IHistoryTxSchema } from '../../schemas';
+
+export const addressHistorySchema = z.discriminatedUnion('success', [
+  z
+    .object({
+      success: z.literal(true),
+      history: IHistoryTxSchema.array(),
+      has_more: z.boolean(),
+      first_hash: z.string().nullish(),
+      first_address: z.string().nullish(),
+    })
+    .passthrough(),
+  z.object({ success: z.literal(false), message: z.string() }).passthrough(),
+]);
+
+export type AddressHistorySchema = z.infer<typeof addressHistorySchema>;
```

Comment:
suggestion(minor): Change naming convention.

`AddressHistorySchema` and `addressHistorySchema` are too similar and may cause confution.
Some options I think can be more indicative:
- `AddressHistoryType` and `AddressHistorySchema`
- `AddressHistory` and `AddressHistorySchema`
- `TAddressHistory` and `AddressHistorySchema`
- `TAddressHistorySchema` and `AddressHistorySchema`

This is also relevant for the other zod schemas, but what do you think?

Reply:
Just to clarify, the schema itself (`addressHistorySchema` in this example) is an object, not a type. So it shouldn't use PascalCase, right? I assumed the casing would be enough to avoid confusion: `addressHistorySchema` is the schema object and `AddressHistorySchema` is the schema type. And TypeScript will make sure they can't be used interchangeably.

But I would be fine with changing the name of the types to include `Type`, for example.

Reply:
While I don't think we should use it here, we usually name a constant for later use with all caps snake case

I do think we should have a consistent naming convention since this validation schema will be used throughout the lib and wallets, so having inconsistent or non-descriptive names may increase overall confusion on the code.
But we can discuss this to refactor on another PR, what do you think?